### PR TITLE
Contacts

### PR DESCRIPTION
**Pull request by or-else**
Login screen is now responsive and scrolls.
Sign up screen allows loading of avatar, scales it down and sends in the {acc} message.

### CODE REVIEW

File: Tinodios/UiUtils.swift [hunk 1]
```diff
@@ -63,4 +63,67 @@ class UiUtils {
         // Reset red border to default.
         field.layer.borderWidth = 0.0
     }
+
+    // Resize image to given dimentions. If aspect ratios are different, clip the central part of the source
+    // image and scale it down to the given dimentions.
+    public static func resizeImage(image imageIn: UIImage?, width: Float, height: Float, clip: Bool) -> UIImage? {
```
Comment (aforge) [inline Tinodios/UiUtils.swift]: Can you make it a UIImage extension?

File: Tinodios/RegisterViewController.swift [hunk 2]
```diff
@@ -26,27 +26,33 @@ class RegisterViewController: UIViewController {
         super.viewDidLoad()
 
         self.imagePicker = ImagePicker(presentationController: self, delegate: self)
+
+        // Listen to text change events to clear the possible error from eralier attempt.
```
[the remark this answers is not in the document]
Comment (or-else) [reply Tinodios/RegisterViewController.swift]: done

File: Tinodios/RegisterViewController.swift [hunk 3]
```diff
@@ -26,27 +26,33 @@ class RegisterViewController: UIViewController {
...
-        guard let name = nameText.text else {
-            return
-        }
-        guard let credential = credentialText.text else {
+        let login = UiUtils.ensureDataInTextField(loginText)
+        let pwd = UiUtils.ensureDataInTextField(pwdText)
+        let name = UiUtils.ensureDataInTextField(nameText)
+        let credential = UiUtils.ensureDataInTextField(credentialText)
+
+        if login == "" || pwd == "" || name == "" || credential == "" {
```
[the remark this answers is not in the document]
Comment (or-else) [reply Tinodios/RegisterViewController.swift]: done

File: Tinodios/UiUtils.swift [hunk 4]
```diff
@@ -63,4 +63,67 @@ class UiUtils {
...
+    // Returns an tuple which contains destination image sizes, source sizes, and offsets
+    // into source (when 'clip' is true).
+    //
+    // The 'clip' parameter forces image to have the new dimensions. Otherwise the
+    // image keeps the original aspect ratio with width and hight being under the
+    // maxWindth/maxHeight
+    public static func fitImageSize(originalWidth: Float, originalHeight: Float, maxWidth: Float, maxHeight: Float, clip: Bool) -> (dstWidth: Float, dstHeight: Float, xOffset: Int, yOffset: Int, srcWidth: Float, srcHeight: Float)? {
+
+        // Sanity check
+        if originalWidth <= 0 || originalHeight <= 0 || maxWidth <= 0 || maxHeight <= 0 {
```
[the remark this answers is not in the document]
Comment (or-else) [reply Tinodios/UiUtils.swift]: done

File: Tinodios/UiUtils.swift [hunk 5]
```diff
@@ -44,4 +44,23 @@ class UiUtils {
         // TODO: logout on failure and route to login view.
         return try? me!.subscribe(set: nil, get: get)
     }
+
+    // Get text from UITextField or mark the field red if the field is blank
+    public static func ensureDataInTextField(_ field: UITextField) -> String {
+        let text = (field.text ?? "").trimmingCharacters(in: .whitespacesAndNewlines)
+        if (text.isEmpty) {
```
[the remark this answers is not in the document]
Comment (or-else) [reply Tinodios/UiUtils.swift]: done

File: Tinodios/UiUtils.swift (hunk 1 above)
Comment (or-else) [reply Tinodios/UiUtils.swift]: done

File: Tinodios/RegisterViewController.swift [hunk 6]
```diff
@@ -49,7 +49,7 @@ class RegisterViewController: UIViewController {
         let name = UiUtils.ensureDataInTextField(nameText)
         let credential = UiUtils.ensureDataInTextField(credentialText)
 
-        guard login == "" || pwd == "" || name == "" || credential == "" else { return }
+        guard login != "" && pwd != "" && name != "" && credential != "" else { return }
```
Comment (aforge) [inline Tinodios/RegisterViewController.swift]: Can you change it to guard !login.isEmpty && !pwd.isEmpty && !name.isEmpty && !credential.isEmpty else { return }

**Pull request merged**